### PR TITLE
naga-cli 0.10.0 (new formula)

### PR DESCRIPTION
**Pull request by pthariensflame**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

### CODE REVIEW

File: Formula/naga-cli.rb [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+    # patch required to fix the version number on the naga-cli crate
+    patch do
+      url "https://github.com/gfx-rs/naga/commit/350171ed2f45612386cc91a0ee9ff79d918939a9.patch?full_index=1"
+      sha256 "45d8d6f0c799cc6ebf198c76dbf991889d4483ba401b8c12b9cdd425a627fa19"
+    end
+  end
+
+  depends_on "rust" => :build
+
+  conflicts_with "naga", because: "naga-cli also ships a naga binary"
```
Comment (carlocab) [inline Formula/naga-cli.rb:20]: We can add the `conflicts_with` after this is merged so we don't need to rebuild `naga` too.

File: Formula/naga-cli.rb [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+
+  stable do
+    url "https://github.com/gfx-rs/naga/archive/refs/tags/v0.9.0.tar.gz"
+    sha256 "5d31b90774b1bb7ef77056911d2a49014ebe4863a6faa83f696f208bbbf3ee02"
+
+    # patch required to fix the version number on the naga-cli crate
+    patch do
+      url "https://github.com/gfx-rs/naga/commit/350171ed2f45612386cc91a0ee9ff79d918939a9.patch?full_index=1"
+      sha256 "45d8d6f0c799cc6ebf198c76dbf991889d4483ba401b8c12b9cdd425a627fa19"
+    end
```
Comment (carlocab) [inline Formula/naga-cli.rb:15]: We'll probably wait until a new version is cut with this patch before merging this.

Comment (cho-m) [reply Formula/naga-cli.rb:15]: New release: https://github.com/gfx-rs/naga/releases/tag/v0.10.0

File: Formula/naga.rb [hunk 3]
```diff
@@ -20,6 +20,8 @@ class Naga < Formula
 
   uses_from_macos "ncurses"
 
+  conflicts_with "naga-cli", because: "naga also ships a naga binary"
+
```
Comment (carlocab) [inline Formula/naga.rb:24]: Let's add the `conflicts_with` separately so we don't have to rebuild these bottles.

**Review by carlocab — CHANGES_REQUESTED** [review]

File: Formula/naga-cli.rb [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+  desc "Shader translation command-line tool"
+  homepage "https://github.com/gfx-rs/naga"
+  url "https://github.com/gfx-rs/naga/archive/refs/tags/v0.10.0.tar.gz"
+  sha256 "9ad07283157af424d130f598be636211188e3ee553bcdf5a9809a40ee28d57c3"
+  license any_of: ["Apache-2.0", "MIT"]
+  head "https://github.com/gfx-rs/naga.git", branch: "master"
+
+  depends_on "rust" => :build
+
+  conflicts_with "naga", because: "naga-cli also ships a naga binary"
```
Comment (pthariensflame) [inline Formula/naga-cli.rb:11]:
```suggestion
```

File: Formula/naga.rb (hunk 3 above)
Comment (pthariensflame) [inline Formula/naga.rb:24]:
```suggestion
```

**Comment by pthariensflame** [comment]
@carlocab Done!

**Review by carlocab — DISMISSED** [review]
Thanks; please open a separate PR to add the `conflicts_with` lines once this is merged.

**Comment by iMichka** [comment]
Squashed everything, and rebased. Let's run a new build to ship this.

**Review by iMichka — APPROVED** [review]

**Comment by iMichka** [comment]
Follow up PR #117886

**Pull request closed without merging**